java/jscip/MessageHandler.java

```diff
@@ -0,0 +1,113 @@
+package jscip;
+
+/** Class representing a message handler (equivalent of SCIP_Messagehdlr). */
+public abstract class MessageHandler
+{
+   private SWIGTYPE_p_SCIP_Messagehdlr _messagehdlrptr; /** pointer address class created by SWIG */
+
+   /** default constructor to construct a Java message handler */
+   protected MessageHandler()
+   {
+      ObjMessagehdlr objmessagehdlr = new ObjMessagehdlr(1) {
+         @Override
+         public void scip_error(SWIGTYPE_p_SCIP_Messagehdlr messagehdlr, SWIGTYPE_p_FILE file, String msg) {
+            error(msg);
+         }
+
+         @Override
+         public void scip_warning(SWIGTYPE_p_SCIP_Messagehdlr messagehdlr, SWIGTYPE_p_FILE file, String msg) {
+            warning(msg);
+         }
+
+         @Override
+         public void scip_dialog(SWIGTYPE_p_SCIP_Messagehdlr messagehdlr, SWIGTYPE_p_FILE file, String msg) {
+            dialog(msg);
+         }
+
+         @Override
+         public void scip_info(SWIGTYPE_p_SCIP_Messagehdlr messagehdlr, SWIGTYPE_p_FILE file, String msg) {
+            info(msg);
+         }
+
+         @Override
+         public SCIP_Retcode scip_free(SWIGTYPE_p_SCIP_Messagehdlr messagehdlr) {
```

Comment:
The class generated by SWIG is called `SCIP_Retcode`, there is no `SCIP_RETCODE` in Java. Java does not have a preprocessor with which I could define a `SCIP_RETCODE` macro. At most, I could rename the generated class, but that would be an API change because Java is case-sensitive. And having both class names is not going to work because the file name must match the class name and there are case-insensitive file systems (FAT, NTFS, exFAT, etc.) and operating systems (Windows) around.

Also note that this API is only the automatically generated low-level `ObjMessageHandler` API. The Java `MessageHandler` is expected to implement the `void close() throws Exception` method instead. We do not use `SCIP_Retcode` in the high-level object-oriented Java API.

Reply:
Thanks, that makes sense!